exercise/static/exercise/user_results.js

```diff
@@ -0,0 +1,102 @@
+			$("#toggle-expand-all-modules").find("span").removeClass("glyphicon-unchecked").addClass("glyphicon-check");
+		}
+
+		// Register an event handler to the "Expand all modules" button
+		$("#toggle-expand-all-modules").on("click", function(event) {
+			const button = $(this);
+			const expandAllModules = localStorage.getItem("expandAllModules") === "true";
+			if (expandAllModules) {
+				$(".module-panel, .module-to-collapse > div").collapse("hide");
+				button.find("span").removeClass("glyphicon-check").addClass("glyphicon-unchecked");
+				localStorage.setItem("expandAllModules", false);
+			} else {
+				$(".module-panel > div").collapse("show");
+				button.find("span").removeClass("glyphicon-unchecked").addClass("glyphicon-check");
+				localStorage.setItem("expandAllModules", true);
+			}
+		});
+	}
+
+	// Check automatic scrolling behavior and show appropriate button
+	const autoScrollBehavior = localStorage.getItem("autoScrollBehavior");
+	if (autoScrollBehavior === "instant") {
+		$("#auto-scroll-behavior-instant").show();
+	} else if (autoScrollBehavior === "off") {
+		$("#auto-scroll-behavior-off").show();
+	} else {
+		$("#auto-scroll-behavior-smooth").show();
+		localStorage.setItem("autoScrollBehavior", "smooth"); // Scroll smoothly by default
+	}
+
+	// Register event handlers to all three scroll behavior buttons
+	$("#cycle-auto-scroll-behavior button").on("click", function(event) {
+		$(this).hide();
+		const autoScrollBehavior = localStorage.getItem("autoScrollBehavior");
+		if (autoScrollBehavior === "instant") {
+			$("#auto-scroll-behavior-off").show();
+			localStorage.setItem("autoScrollBehavior", "off");
+		} else if (autoScrollBehavior === "off") {
+			$("#auto-scroll-behavior-smooth").show();
+			localStorage.setItem("autoScrollBehavior", "smooth");
+		} else {
+			$("#auto-scroll-behavior-instant").show();
+			localStorage.setItem("autoScrollBehavior", "instant");
+		}
+	});
+
+	$('.filter-categories button').on("click", function(event) {
+		var button = $(this);
+		var id = button.attr("data-category");
+		if (button.hasClass("active")) {
+			button.removeClass("active").find("span").removeClass("glyphicon-unchecked").addClass("glyphicon-check");
+			$('.module-panel tr[data-category="' + id + '"]').removeClass("hide");
+		} else {
+			button.addClass("active").find("span").removeClass("glyphicon-check").addClass("glyphicon-unchecked");
+			$('.module-panel tr[data-category="' + id + '"]').addClass("hide");
+		}
+		$('.module-panel').each(function(index, panel) {
+			var mod = $(panel);
+			if (mod.find("tr:not(.hide)").length > 0) {
+				mod.removeClass("hide");
+			} else {
+				mod.addClass("hide");
+			}
+		});
+	});
+
+	// Get current course news from the document and previously shown course news from local storage
+	const news = [];
+	$(".news-panel > .list-group > .list-group-item > .list-group-item-text").each(function() {
+		news.push($(this)[0].innerText);
```

Comment:
This only considers the body of the news item but not the title. I think it would be better to also include the title. For example, the title may have been edited or the news item only has a title and no body.

Reply:
Fixed!